src/hotspot/cpu/riscv/vm_version_riscv.cpp

```diff
@@ -157,12 +157,20 @@ void VM_Version::initialize() {
     FLAG_SET_DEFAULT(UseSHA1Intrinsics, false);
   }
 
-  if (UseSHA256Intrinsics) {
+  if (UseZvknha && UseZvkb) {
```

Comment:
A simple question here: Does the existence of `Zvknhb` means availability of `Zvknha`? Or should this be something like `if ((UseZvknha || UseZvknhb) && UseZvkb)`?

Reply:
It seems like the correct answer is:
`• Zvknhb supports SHA-256 and SHA-512.`

I suggest we start with supporting Zvkn, which is:
Zvkned, Zvknhb, Zvkb, Zvkt
They require: Zve64x

Reply:
Yeah, I agree it's more reasonable to check for `Zvkn` here which stands for NIST Algorithm Suite.
I see the spec says:
```
The Zvknhb and Zvbc Vector Crypto Extensions --and accordingly the composite extensions Zvkn
and Zvks-- require a Zve64x base, or application ("V") base Vector Extension.
```
My understanding is that either `Zve64x` or RVV for our case will do.
So we might want to do this check: `if (UseRVV && UseZvkn)`.

Reply:
Please add new comments in updated code.